obspy/signal/polarization.py

```diff
@@ -86,48 +87,482 @@ def eigval(datax, datay, dataz, fk, normf=1):
+                             plan, ellip]))
+        offset += 1
+
+    return res
+
+
+def particle_motion_odr(stream, noise_thres=0):
+    """
+    Computes the orientation of the particle motion vector based on an
+    orthogonal regression algorithm.
+
+    :param stream: ZNE sorted trace data
+    :type stream: :class:`~obspy.core.stream.Stream`
+    :param noise_tres: variance of the noise sphere; data points are excluded
+        when falling within the sphere with radius sqrt(noise_thres)
+    :type noise_thres: float
+    :returns: azimuth, incidence, error of azimuth, error of incidence
+    """
+    Z = []
+    N = []
+    E = []
+    comp, npts = np.shape(stream)
+
+    for i in range(0, npts):
+        if (stream[0][i] ** 2 + stream[1][i] ** 2 + stream[2][i] ** 2) \
+                > noise_thres:
+            Z.append(stream[0][i])
+            N.append(stream[1][i])
+            E.append(stream[2][i])
+
+    def fit_func(beta, x):
+        # Eventually this is correct: return beta[0] * x + beta[1]
+        return beta[0] * x
+
+    data = scipy.odr.Data(E, N)
+    model = scipy.odr.Model(fit_func)
+    odr = scipy.odr.ODR(data, model, beta0=[1.])
+    out = odr.run()
+    az_slope = out.beta[0]
+    az_error = out.sd_beta[0]
+
+    N = np.asarray(N)
+    E = np.asarray(E)
+    Z = np.asarray(Z)
+    R = np.sqrt(N ** 2 + E ** 2)
+
+    data = scipy.odr.Data(R, np.sqrt(Z ** 2))
+    model = scipy.odr.Model(fit_func)
+    odr = scipy.odr.ODR(data, model, beta0=[1.0])
+    out = odr.run()
+    in_slope = out.beta[0]
+    in_error = out.sd_beta[0]
+
+    azim = math.atan2(1.0, az_slope)
+    inc = math.atan2(1.0, in_slope)
+    az_error = 1.0 / ((1.0 ** 2 + az_slope ** 2) * azim) * az_error
+    in_error = 1.0 / ((1.0 ** 2 + in_slope ** 2) * inc) * in_error
+
+    return math.degrees(azim), math.degrees(inc), az_error, in_error
+
+
+def get_s_point(stream, stime, etime):
+    """
+    Function for computing the trace dependent start time in samples
+
+    :param stime: time to start
+    :type stime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :param etime: time to end
+    :type etime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :returns: spoint, epoint
+    """
+    slatest = stream[0].stats.starttime
+    eearliest = stream[0].stats.endtime
+    for tr in stream:
+        if tr.stats.starttime >= slatest:
+            slatest = tr.stats.starttime
+        if tr.stats.endtime <= eearliest:
+            eearliest = tr.stats.endtime
+
+    nostat = len(stream)
+    spoint = np.empty(nostat, dtype=np.int32, order="C")
+    epoint = np.empty(nostat, dtype=np.int32, order="C")
+    # now we have to adjust to the beginning of real start time
+    if slatest > stime:
+        msg = "Specified start-time is smaller than starttime in stream"
+        raise ValueError(msg)
+    if eearliest < etime:
+        msg = "Specified end-time bigger is than endtime in stream"
```

Comment:
s/end-time/end time/
s/bigger/later/
s/endtime/end time/ (and maybe add in earliest as well)